doc/modules/linear_model.rst

```diff
@@ -888,6 +888,146 @@ to warm-starting (see :term:`Glossary <warm_start>`).
     .. [9] `"Performance Evaluation of Lbfgs vs other solvers"
             <http://www.fuzihao.org/blog/2016/01/16/Comparison-of-Gradient-Descent-Stochastic-Gradient-Descent-and-L-BFGS/>`_
 
+.. _Generalized_linear_regression:
+
+Generalized Linear Regression
+=============================
+
+Generalized Linear Models (GLM) extend linear models in two ways
```

Comment:
What about reversing the order: Call the whole section "Generalized Linear Regression" and this subsection on GLMs "Generalized Linear Models"?

Reply:
Why? I think it's more natural to talk about logistic regression above as a generalized linear model than as generalized linear regression, in particular given our interface that doesn't allow fitting probabilities.

Reply:
> This subsection seems to be specifically about GLMs, while the title of the whole chapter is also "Generalized Linear Models".

Yeah, I'm not sure that inverting the section and subsection title would help.

On one side I understand that if one were to give a title to a section that contains both Ridge and Logistic Regression, GLM could be it. On the other however, the term GLM does have a very specific meaning, and prior to this PR it was never defined (or mentioned for that meter beside of the title) which I also find somewhat misleading.

 Since I wasn't sure how to improve that, I have kept the added "Generalized Linear Regression" subsection.